### PR TITLE
mysql: update to 8.0.24

### LINKED ISSUE #1626: mysql: upgrade to v8 for nextcloud v21
A warning on 20.0.6 states:

> MySQL version "5.7.33" is used. Nextcloud 21 will no longer support this version and requires MySQL 8 or higher.

We'll need to do that. This will be done in a manner that is completely transparent to users, so if you're coming across the issue as a user, don't worry-- you can safely ignore that message.

### PR DESCRIPTION
**Pull request by kyrofa**
This PR resolves #1626 by upgrading MySQL to the latest v8 release, thus paving the path to Nextcloud v21, which requires this.

### CODE REVIEW

**Comment by kyrofa** [comment]
A build of this PR for every architecture is available on the `beta/pr-1710` channel. Please help test it!

    $ sudo snap install nextcloud --channel=beta/pr-1710

Or, if you already have it installed:

    $ sudo snap refresh nextcloud --channel=latest/beta/pr-1710

**Comment by kyrofa** [comment]
One more night to take this for a spin, I'll be landing it first thing in my morning assuming CI is green.

File: snap/snapcraft.yaml [hunk 1]
```diff
@@ -325,84 +325,77 @@ parts:
...
+      - -DWITH_DEBUG=OFF
+      - -DWITH_INNODB_EXTRA_DEBUG=OFF
       - -DINSTALL_MYSQLTESTDIR=
+      - -DDOWNLOAD_BOOST=OFF
+      - -DWITH_BOOST=$SNAPCRAFT_STAGE
+
+    override-build: |
+      snapcraftctl build
+      # MySQL v8 is massive. Strip it.
+      find "$SNAPCRAFT_PART_INSTALL/bin" -type f -exec sh -c 'grep -IL . "$1" || strip --strip-all "$1"' sh "{}" \;
```
Comment (pachulo) [inline snap/snapcraft.yaml:365]: Is this sole `strip` the cause of the huge slimming of the snap?
```
channels:
  latest/stable:    20.0.9snap1              2021-04-27 (27441) 287MB -
  latest/candidate: 20.0.9snap2              2021-05-08 (27703) 219MB -
```

Comment (kyrofa) [reply snap/snapcraft.yaml:365]: Yeah, you like that? Without it, it would have been over 500MBs :scream: .

**Pull request merged**